katana/grib_crop_wgrib2.py

```diff
@@ -1,15 +1,54 @@
 import datetime
+import glob
 import os
 from subprocess import PIPE, Popen
 
 import dateparser
 import numpy as np
 import utm
 
+fmt1 = '%Y%m%d'
+fmt2 = '%H'
 
-def grib_to_sgrib(fp_in, out_dir, file_dt, x, y, logger,
-                  buff=1500, zone_letter='N', zone_number=11,
-                  nthreads_w=1):
+
+def call_wgrib2(action, logger):
+    # run command line using Popen
+    logger.debug('Running "{}"'.format(action))
+    s = Popen(action, shell=True, stdout=PIPE,
+              stderr=PIPE, universal_newlines=True)
+
+    # stream the output of WindNinja to the logger
+    output = []
+    fatl = False
+    with s.stdout:
+        for line in iter(s.stdout.readline, ""):
+            logger.debug(line.rstrip())
+            output.append(line.rstrip())
+
+    with s.stderr:
+        # eline = s.stderr.readline()
+        for line in iter(s.stderr.readline, ""):
+            logger.warning(line.rstrip())
+            output.append(line.rstrip())
+
+            # if it failed, find a different forecast hour
+            if "FATAL" in line:
```

Comment:
There could be other errors to catch here. For instance, 
$ WindNinja_cli
with no arguments gives:

ERROR 6: GDAL/OGR not compiled with libcurl support, remote requests not supported.
Allowed options:

Reply:
Katana will never see that error as it's always providing a config file. However, in this case the `return_code = s.wait()` will not succeed and raise an error. Plus this is for `wgrib2` and not the `WindNinja_cli`